internal/pkg/agent/application/fleet_server_bootstrap.go

```diff
@@ -104,6 +110,7 @@ func InjectFleetConfigComponentModifier(fleetCfg *configuration.FleetAgentConfig
 						if v, ok := unitCfgMap["fleet"]; ok {
 							if m, ok := v.(map[string]interface{}); ok {
 								m["host"] = cfg["host"]
+								m["hosts"] = fleetHosts
```

Comment:
this will always replace "hosts" content, even if for the cases where it is defined, right? is this intentional?

Reply:
yeah i considered check but i would need to check for length as well, in case it has hosts but it's empty array. this is cleaner and should not matter.
if you have perf concerns i can add check

Reply:
you mentioned that there was an existing  use case where the "hosts" property was already somehow specified (by the user?) do we want to overwrite this completely?

Reply:
i meant in original fleet configuration, the one we are injecting here. 
the result of this change should not be data loss. 
we either replace empty hosts with []{host} or list of hosts with the same list.